test/jsunit/parser/parser.test.js

```diff
@@ -14,115 +14,134 @@ describe('Parser catroid program tests', () => {
-      const catXml = parser.convertProgramString(xmlString);
-
-      return (catXml === undefined);
+      try {
+        const xmlString = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?><program><header><catrobatLanguageVersion>0.993</catrobatLanguageVersion></header><scenes><scene><name>игра</name><objectList></objectList></scene></scenes></program>`;
+        parser.convertProgramToJSONDebug(xmlString);
+      }
+      catch (e) {
+        if (e.message === 'Found program version 0.993, minimum supported is 0.994')
+          return true;
+      }
+      return false;
     })).toBeTruthy();
   });
 
   test('Recognizes supported program version', async () => {
     expect(await page.evaluate(() => {
       const xmlString = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?><program><header><catrobatLanguageVersion>0.994</catrobatLanguageVersion></header><scenes><scene><name>игра</name><objectList></objectList></scene></scenes></program>`;
-      const catXml = parser.convertProgramString(xmlString);
-
-      return (catXml instanceof XMLDocument);
+      const programJSON = parser.convertProgramToJSONDebug(xmlString);
+      return (programJSON !== undefined && programJSON.scenes !== undefined &&
+        programJSON.scenes[0].name !== undefined && programJSON.scenes[0].objectList !== undefined);
     })).toBeTruthy();
   });
 
   test('Handle empty program properly', async () => {
     expect(await page.evaluate(() => {
       const xmlString = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?><program><header><catrobatLanguageVersion>0.994</catrobatLanguageVersion></header><scenes></scenes></program>`;
-      const catXml = parser.convertProgramString(xmlString);
-
-      return (catXml instanceof XMLDocument
-        && catXml.firstChild.tagName === 'xml'
-        && catXml.firstChild.childElementCount === 0);
+      const programJSON = parser.convertProgramToJSONDebug(xmlString);
+      if (programJSON === undefined)
+        return false;
+      return (programJSON.scenes !== undefined);
     })).toBeTruthy();
   });
 
   test('Handle empty single scene properly', async () => {
     expect(await page.evaluate(() => {
       const xmlString = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?><program><header><catrobatLanguageVersion>0.994</catrobatLanguageVersion></header><scenes><scene><name>tscene</name><objectList></objectList></scene></scenes></program>`;
-      const catXml = parser.convertProgramString(xmlString);
-
-      return (catXml instanceof XMLDocument
-        && catXml.firstChild.tagName === 'xml'
-        && catXml.getElementsByTagName('scene').length === 1
-        && catXml.getElementsByTagName('scene')[0].getAttribute('type') === 'tscene');
+      const programJSON = parser.convertProgramToJSONDebug(xmlString);
```

Comment:
fixed all parser tests by using `convertProgramToJSONDebug` instead of `convertProgramString`.